TCMSApp/src/client/app/tests/tests-list.controller.js

```diff
@@ -99,5 +142,266 @@
                 controllerAs: 'vmSuite'
             });
         }
+
+        function setFilterQueryWatcher() {
+            $scope.$watch('filterQuery', function (newVal, oldVal) {
+                if (oldVal !== undefined) {
+                    refreshFilteredTests();
+                }
+            });
+        }
+
+        function setCurrentTests() {
+            vm.tests.map(function (item, i, array) {
+                if (item.suite === vm.currentSuite._id) {
+                    vm.currentTests.push(item);
+                }
+            });
+        }
+
+        function refreshFilteredTests() {
+            vm.filteredTests.length = 0;
+
+            $filter('smartFilter')(vm.currentTests, $scope.filterQuery, vm.filterFields)
+                .map(function (item, index, arr) {
+                    vm.filteredTests.push(item);
+                });
+
+            if (vm.selectedTests.length) {
+                setCheckAllCheckbox();
+            }
+        }
+
+        //functions for checkboxes -------------------------------------------------
+
+        function setIndeterminateById(id, value) {
```

Comment:
i suppose that several of these functions are private (used only here, not exposed to vm). Let's name they like `_funcName`